pkg/mutation/mutators/assignimage/imageparser.go

```diff
@@ -0,0 +1,114 @@
+package assignimage
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+type image struct {
+	domain string
+	path   string
+	tag    string
+}
+
+var (
+	// domainRegexp defines a schema for a domain component. Primarily, this is to
```

Comment:
It's worth noting that a lot of this schema validation is done in code and not via regex (looking at the regex in isolation made me think this was insufficient)